test/e2e/kvc_test.go

```diff
@@ -0,0 +1,370 @@
+					Capacity:    "5Gi",
+					Labels:      map[string]string{},
+					Options: map[string]string{
+						"awsCredentialsSecretName": "s3-creds",
+					},
+				},
+			},
+			expSuccess: false,
+			expError:   fmt.Sprintf("labels cannot be empty"),
+			expHP:      false,
+			expNA:      false,
+			expPVC:     false,
+		},
+		{
+			description: "single vc - S3 - no creds error",
+			volumeConfigs: []crv1.VolumeConfig{
+				{
+					ID:          "vol1",
+					Replicas:    1,
+					SourceType:  "S3",
+					SourceURL:   "s3://e2e-test/cifar-100-python.tar.gz",
+					EndpointURL: "http://10.63.255.0:9000",
+					AccessMode:  "ReadWriteOnce",
+					Capacity:    "5Gi",
+					Labels: map[string]string{
+						"key1": "val1",
+						"key2": "val2",
+					},
+					Options: map[string]string{},
+				},
+			},
+			expSuccess: false,
+			expError:   fmt.Sprintf("awsCredentialsSecretName key has to be set in options"),
+			expHP:      false,
+			expNA:      false,
+			expPVC:     false,
+		},
+		{
+			description: "single vc - NFS - no label error",
+			volumeConfigs: []crv1.VolumeConfig{
+				{
+					ID:         "vol1",
+					SourceType: "NFS",
+					AccessMode: "ReadWriteMany",
+					Capacity:   ".5Gi",
+					Labels:     map[string]string{},
+					Options: map[string]string{
+						"server": "10.63.249.182",
+						"path":   "/",
+					},
+				},
+			},
+			expSuccess: false,
+			expError:   fmt.Sprintf("labels cannot be empty"),
+			expHP:      false,
+			expNA:      false,
+			expPVC:     false,
+		},
+		{
+			description: "single vc - NFS - no server in options error",
+			volumeConfigs: []crv1.VolumeConfig{
+				{
+					ID:         "vol1",
+					SourceType: "NFS",
+					AccessMode: "ReadWriteMany",
+					Capacity:   ".5Gi",
+					Labels: map[string]string{
+						"key1": "val1",
+						"key2": "val2",
+					},
+					Options: map[string]string{
+						"path": "/",
+					},
+				},
+			},
+			expSuccess: false,
+			expError:   fmt.Sprintf("server has to be set in options"),
+			expHP:      false,
+			expNA:      false,
+			expPVC:     false,
+		},
+		{
+			description: "single vc - NFS - no path in options error",
+			volumeConfigs: []crv1.VolumeConfig{
+				{
+					ID:         "vol1",
+					SourceType: "NFS",
+					AccessMode: "ReadWriteMany",
+					Capacity:   ".5Gi",
+					Labels: map[string]string{
+						"key1": "val1",
+						"key2": "val2",
+					},
+					Options: map[string]string{
+						"server": "10.63.249.182",
+					},
+				},
+			},
+			expSuccess: false,
+			expError:   fmt.Sprintf("path has to be set in options"),
+			expHP:      false,
+			expNA:      false,
+			expPVC:     false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		volman := makeVolumeManager(testCase.volumeConfigs)
+		createdVolman, err := crdClient.Create(volman)
+		require.Nil(t, err)
+		/*
+			defer func() {
+				delOpts := &metav1.DeleteOptions{}
+				crdClient.Delete(volman.GetName(), delOpts)
+			}()
+		*/
+
+		if testCase.expSuccess {
```

Comment:
Here I think we should check if there exists a hostpath/NA/PVC for a particular id. Like we should check if hostpath and NA are created for s3 source type, and PVC is created for nfs type. We should add a TODO here too and add it in as a enhancement to the e2e tests.

Reply:
Also we should check for PVs along with PVCs.

Reply:
I added a TODO for this. Will create issues.